vector/src/main/java/im/vector/app/features/form/FormSwitchItem.kt

```diff
@@ -61,10 +61,15 @@ abstract class FormSwitchItem : VectorEpoxyModel<FormSwitchItem.Holder>() {
 
         holder.switchView.isEnabled = enabled
 
-        holder.switchView.setOnCheckedChangeListener(null)
-        holder.switchView.isChecked = switchChecked
-        holder.switchView.setOnCheckedChangeListener { _, isChecked ->
-            listener?.invoke(isChecked)
+        if (holder.view.isAttachedToWindow) {
+            // the view is attached to the window
+            // So it is a rebind of new data and you could ignore it assuming this is value that was already inputted into the view.
+        } else {
+            holder.switchView.setOnCheckedChangeListener(null)
+            holder.switchView.isChecked = switchChecked
+            holder.switchView.setOnCheckedChangeListener { _, isChecked ->
+                listener?.invoke(isChecked)
+            }
```

Comment:
Create another extension for this case?

Reply:
when i'll have two :D ?

Reply:
Ok, but the idea is to improve our framework :)